app/models/piece.rb

```diff
@@ -82,4 +66,21 @@ def move_to!(new_row, new_column)
       end
     end
   end
+
+  def moving_into_check?(new_row, new_column)
+    #We will switch to the new coordinates temporarily and see if we are in check!
+    current_row = self.position_row
+    current_column = self.position_column
+    self.update_attributes(position_row: new_row, position_column: new_column)
+    #Get the status of check?
```

Comment:
Sure so it turns out we need two return values from check?

The syntax to return multiple things is the check method is below: 
return color, piece
And to get these values in the moving_into_check? method we use the syntax below: 
status, culprit = self.game.check?

In status we store the color of the player in check. If no one is in check, this returns nil.
In culprit, I store the piece that is causing us to be in check.

I need to store the 'culprit' because if our piece is moving to the spot where the culprit is located, we are not putting ourselves in check. We are removing the threat. It is a special case of moving_into_check?